core/src/main/java/io/parsingdata/metal/token/Sub.java

```diff
@@ -40,17 +40,38 @@ public Sub(final Token op, final ValueExpression addr, final Encoding enc) {
 
     @Override
     protected ParseResult parseImpl(final String scope, final Environment env, final Encoding enc) throws IOException {
-        final OptionalValue ov = _addr.eval(env, enc);
-        if (!ov.isPresent()) { return new ParseResult(false, env); }
-        final long ref = ov.get().asNumeric().longValue();
-        if (env.order.hasGraphAtRef(ref)) { return new ParseResult(true, new Environment(env.order.add(new ParseRef(ref, this)), env.input, env.offset)); }
-        final ParseResult res = _op.parse(scope, new Environment(env.order.addBranch(this), env.input, ref), enc);
+        final OptionalValueList addrs = _addr.eval(env, enc);
+        if (addrs.isEmpty() || addrs.containsEmpty()) { return new ParseResult(false, env); }
+        final ParseResult res = iterate(scope, addrs, new Environment(env.order.addBranch(this), env.input, env.offset), enc);
```

Comment:
Ah, so here is the single `addBranch` (and not deeper into the iteration below).